design/split-image.md

```diff
@@ -0,0 +1,40 @@
+# Decoupling Rook Agent from Backend Images
+
+## Tl;dr
+
+Rook operator should allow `rook` binary in its own container image while running backend storage (currently just Ceph) in a different image.
+
+## Current Flow
+
+Currently `rook` and all Ceph daemons and binaries are packaged in the same container image. 
+
+The following illustrates the flow of starting `ceph-mon`:
+
+
+| Seq  | Rook Operator  | Rook mon agent  | Ceph Daemon  |
+|---|---|---|---|
+| 1  | Create Replicaset to run `rook mon` command using the same container image   |   |   |
+| 2  |   | `rook mon` exec `ceph-mon`  |   |
+| 3  |   |   |  `ceph-mon` starts |
+
+
+## Issues
+
+Having rook and ceph in the same container image has the following issues:
+
+- Unwanted ceph daemon restart during upgrade. Whenever rook binary is upgraded, a new image is created and the replicaset is refreshed. At this time, ceph daemons are restarted as well, even ceph binaries are the same. This tight coupling creates more disruptions when other storage backends are supported.
+
+- Harder to use other ceph images. If customers choose to run ceph in a different image, e.g. `docker.io/ceph/daemon`, they have to make a new image to pick up `rook` binary.
+
+## New Flow
+
+| Seq  | Rook Operator  | Rook mon agent  | Ceph Daemon  |
+|---|---|---|---|
+| 1  | Create Replicaset to run `rook mon` command using the same container image in `init container` and use `docker.io/ceph/daemon` to run `ceph-mon` command in main container   |   |   |
```

Comment:
Running the daemon container would be a concern since it would be running a different version. To keep the versioning story simpler, running the rook container for all the daemons is needed. It would be good to talk through this scenario more.